File: Backtracking/NM3/Java_NM3/src/NM3.java
import java.io.*;
import java.util.StringTokenizer;

public class NM3 {
    static final int MAX = 7 + 1;

    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int n, m;
    static int[] numList = new int[MAX];

    static void permutationWithRepetition(int cnt) throws IOException {
        if (cnt == m) {
            for (int i = 0; i < m; i++) {
                bw.write(numList[i] + " ");
            }
            bw.newLine();
            return;
        }

        for (int i = 1; i <= n; i++) {
            numList[cnt] = i;
            permutationWithRepetition(cnt + 1);
        }
    }

    public static void main(String[] args) throws IOException {
        StringTokenizer st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());

        permutationWithRepetition(0);

        bw.flush();
    }
}

File: Backtracking/NM3/CPP_NM3/NM3.cpp
#include <iostream>
#include <vector>

using namespace std;

int n, m;
vector<int> num_list;

void permutation_with_repetition(int cnt)
{
    if (cnt == m)
    {
        for (int i = 0; i < num_list.size(); i++)
        {
            cout << num_list[i] << " ";
        }
        cout << "\n";
        return;
    }

    for (int i = 1; i <= n; i++)
    {
        num_list.push_back(i);
        permutation_with_repetition(cnt + 1);
        num_list.pop_back();
    }
}

int main()
{
    cin >> n >> m;

    permutation_with_repetition(0);

    return 0;
}
